modu/source_repository/source_repository.py

```diff
@@ -10,7 +10,7 @@
 import chardet
 
 _TREESITTER_BUILD: Path = Path("build/my-languages.so")
-
+_IGNORE_FILE_PATH: Path = Path("modu/util/.moduignore")
```

Comment:
I think it is important, that the ignore file's location can be passed as a command line argument so that the user has more control over ignoring different files for different pieces of software that they are analyzing

Reply:
will do

Reply:
This change will also require changes in all source reposistory constructors, so i think we should do this in a seperate pr

Reply:
We can still default it to `modu/util/.moduignore` using a keyword based parameter in python kinda like how when you call `open()` it will assume utf-8, but if you call `open(encoding=utf-16)` you override it. In that way we also don't need to change a bunch of tests.